File: tippspiel2-spring/src/main/kotlin/com/github/zemke/tippspiel2/persistence/model/Standing.kt
package com.github.zemke.tippspiel2.persistence.model

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@Entity
data class Standing(

    @Id @GeneratedValue val id: Long?,
    @Min(0) @NotNull var points: Int = 0,
    @Min(0) @NotNull var exactBets: Int = 0,
    @Min(0) @NotNull var goalDifferenceBets: Int = 0,
    @Min(0) @NotNull var winnerBets: Int = 0,
    @Min(0) @NotNull var wrongBets: Int = 0,
    @Min(0) @NotNull var missedBets: Int = 0,
    @ManyToOne(optional = false) val user: User,
    @ManyToOne(optional = false) val bettingGame: BettingGame
) {

    fun reset(): Standing {
        this.points = 0
        this.exactBets = 0
        this.goalDifferenceBets = 0
        this.winnerBets = 0
        this.wrongBets = 0
        this.missedBets = 0
        return this
    }
}

File: tippspiel2-ember/src/main/webapp/app/models/standing.js
import DS from 'ember-data';

export default DS.Model.extend({
  points: DS.attr('number'),
  exactBets: DS.attr('number'),
  goalDifferenceBets: DS.attr('number'),
  winnerBets: DS.attr('number'),
  wrongBets: DS.attr('number'),
  missedBets: DS.attr('number'),
  user: DS.belongsTo('user'),
  bettingGame: DS.belongsTo('betting-game'),
});
